### ISSUE TITLE
undefined: channel.OsChannel

### ISSUE THREAD

**Issue by lomoonmoonbird** [chaosblade-io/chaosblade#1016]
## Issue Description

编译chaosblade，执行命令  make build_linux

Type: *bug report* , *feature request* or *question*

### Describe what happened (or what feature you want)
git clone -b master https://github.com/lomoonmoonbird/chaosblade-exec-cplus.git target/cache/chaosblade-exec-cplus
Cloning into 'target/cache/chaosblade-exec-cplus'...
make -C target/cache/chaosblade-exec-cplus
make[1]: Entering directory '/go/src/github.com/lomoonmoonbird/chaosblade/target/cache/chaosblade-exec-cplus'
rm -rf build-target/chaosblade-1.7.3
mkdir -p build-target/chaosblade-1.7.3/yaml build-target/chaosblade-1.7.3/lib/cplus/script
env CGO_ENABLED=1 GO111MODULE=on go build -ldflags="-linkmode external -extldflags -static" -o build-target/chaosblade-1.7.3/lib/cplus/chaosblade-exec-cplus main.go
go: downloading github.com/sirupsen/logrus v1.4.2
go: downloading github.com/shirou/gopsutil v2.20.5+incompatible
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading golang.org/x/sys v0.0.0-20190916202348-b4ddaad3f8a3
# github.com/lomoonmoonbird/chaosblade-exec-cplus/common
common/channel.go:27:15: undefined: channel.OsChannel
make[1]: *** [Makefile:40: build_cplus] Error 2
make[1]: Leaving directory '/go/src/github.com/lomoonmoonbird/chaosblade/target/cache/chaosblade-exec-cplus'
make: *** [Makefile:208: cplus] Error 2
make[1]: *** [Makefile:277: build_linux_with_arg] Error 2
make[1]: Leaving directory '/home/chaosblade/chaosblade'
make: *** [Makefile:111: build_linux] Error 2

### How to reproduce it (as minimally and precisely as possible)

1. make build_linux
2. 
3.

**Comment by zhanluxianshen** [comment]
Your clone repo is not ```https://github.com/chaosblade-io/chaosblade``` .

**Comment by lomoonmoonbird** [comment]
> Your clone repo is not `https://github.com/chaosblade-io/chaosblade` .

yes, my clone repo is definitely is https://github.com/chaosblade-io/chaosblade,
![image](https://github.com/chaosblade-io/chaosblade/assets/10216631/2aee598a-80ff-4329-9847-32ed8a874c6e)

in the chaosblade dirctory, it is the same as chaosblade-io repo:
![image](https://github.com/chaosblade-io/chaosblade/assets/10216631/cee93a73-7383-4f6c-9087-59a43952bfaf)

so in this direcotry i ran "make build_Linux" command, and then when built on exec_cplus step, it throw an errors

**Comment by oss-sentinel-ai** [comment]
**Issue Evaluation**

Category: `type/bug` | Status: **Resolved** | Severity: **Low**

**Problem:**
Build fails with `undefined: channel.OsChannel` error when compiling `chaosblade-exec-cplus`.

**Root Cause:**
The user cloned a **forked repository** (`lomoonmoonbird/chaosblade-exec-cplus`) instead of the official repository (`chaosblade-io/chaosblade-exec-cplus`). The fork is outdated and missing recent changes.

**Solution (from maintainer @zhanluxianshen):**
Use the official repository:
```bash
# Clone the official repo
git clone https://github.com/chaosblade-io/chaosblade.git
cd chaosblade

# Build
make build_linux
```

**Additional Notes:**
- Always clone from the official `chaosblade-io` organization
- Forked repositories may be outdated or have divergent changes
- Check the `go.mod` file to ensure dependencies point to official repos

**Recommendation:** This is a user error, not a ChaosBlade bug. Users should always use official repositories for building.

---
*Automated evaluation by [oss-taishan-ai](https://github.com/oss-taishan-ai) - ChaosBlade Community Bot*
